sink/kafka.go

```diff
@@ -0,0 +1,114 @@
+package sink
+
+import (
+	"fmt"
+	"github.com/Shopify/sarama"
+	log "github.com/Sirupsen/logrus"
+	"os"
+	"strings"
+	"time"
+)
+
+// KafkaSink ...
+type KafkaSink struct {
+	// Kafka brokers to send metrics to
+	Brokers []string
+	// Kafka topic
+	Topic string
+
+	producer sarama.SyncProducer
+
+	stopCh chan interface{}
+	putCh  chan []byte
+}
+
+// NewKafka ...
+func NewKafka() (*KafkaSink, error) {
+
+	brokers := os.Getenv("SINK_KAFKA_BROKERS")
+	if brokers == "" {
+		return nil, fmt.Errorf("[sink/kafka] Missing SINK_KAFKA_BROKERS")
+	}
+
+	brokerList := strings.Split(brokers, ",")
+	log.Debugf("[sink/kafka] Kafka brokers: %s", strings.Join(brokerList, ", "))
+
+	topic := os.Getenv("SINK_KAFKA_TOPIC")
+	if topic == "" {
+		return nil, fmt.Errorf("[sink/kafka] Missing SINK_KAFKA_TOPIC")
+	}
+	log.Debugf("[sink/kafka] Kafka topic: %s", topic)
+
+	config := sarama.NewConfig()
+	config.Producer.Return.Successes = true
+
+	producer, err := sarama.NewSyncProducer(brokerList, config)
+	if err != nil {
+		log.Fatal(err)
+		os.Exit(1)
+	}
+
+	return &KafkaSink{
+		Brokers:  brokerList,
+		Topic:    topic,
+		producer: producer,
+		stopCh:   make(chan interface{}),
+		putCh:    make(chan []byte, 1000),
+	}, nil
+}
+
+// Start ...
+func (s *KafkaSink) Start() error {
+	// Stop chan for all tasks to depend on
+	s.stopCh = make(chan interface{})
+
+	go s.write()
+
+	// wait forever for a stop signal to happen
+	for {
+		select {
+		case <-s.stopCh:
+			break
```

Comment:
you can just `return nil` here instead and drop the double `break`

Reply:
copy pasted from kinesis =)

Reply:
I like how my Go code skills has improved since I did the original code then :P

Reply:
as you can see, i'm writing it for my  Go skills improve too! =)